charts/kommander-operator/templates/pre_upgrade_kommandercore_hook.yaml

```diff
@@ -0,0 +1,86 @@
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ .Chart.Name }}-pre-upgrade
+  namespace: {{ .Release.Namespace }}
+  annotations:
+    "helm.sh/hook": pre-upgrade
+    "helm.sh/hook-weight": "-5"
+    "helm.sh/hook-delete-policy": before-hook-creation
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: {{ .Chart.Name }}-pre-upgrade
+  annotations:
+    "helm.sh/hook": pre-upgrade
+    "helm.sh/hook-weight": "-5"
+    "helm.sh/hook-delete-policy": before-hook-creation
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: cluster-admin
+subjects:
+  - kind: ServiceAccount
+    name: {{ .Chart.Name }}-pre-upgrade
+    namespace: {{ .Release.Namespace }}
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ .Chart.Name }}-pre-upgrade
+  namespace: {{ .Release.Namespace }}
+  annotations:
+    "helm.sh/hook": pre-upgrade
+    "helm.sh/hook-weight": "-5"
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded
+data:
+  pre_process_kcore.sh: |-
+    #!/bin/bash
+    set -eo pipefail
+
+    # get the version of kommander before upgrade
+    KOMMANDER_VERSION_PREUPGRADE=$(kubectl get helmrelease -n kommander kommander -o jsonpath='{.status.lastAppliedRevision}')
+    cat <<EOF > kcore_patch.yaml
+    apiVersion: dkp.d2iq.io/v1alpha1
+    kind: KommanderCore
+    metadata:
+      name: kommander-core
+    status:
+      version: $KOMMANDER_VERSION_PREUPGRADE
+    EOF
+
+    kubectl patch kommandercore kommander-core --subresource='status' --type='merge' --patch-file kcore_patch.yaml
```

Comment:
should we check first that the status version is not already populated? I guess I'm thinking about if this script gets rerun after kommandercore has been upgraded, if it might overwrite it to the previous version again. Like with config updates for example if that triggers a HR "upgrade" action

Reply:
That's a good point! But even with that I think a helm upgrade after kommander upgrade will not make it patch to the previous version cuz at that point kommander HR is already at the upgraded version, thus this patch will result in no changes.

Reply:
I see what you mean, that's true.. I guess I just don't feel so comfortable with this hook potentially writing that status after the first initial state, because it should really be solely controlled by the operator. There is a possibility that before upgrade succeeds in the operator, this script is rerun and prematurely updates the status to the current ("new") version, in which case it is possible that the operator mid-upgrade process think that upgrade is done and won't continue on? It's probably not very likely, but I think it should be easy to avoid this by adding a check here just to limit the updating of these resources to the operator besides this one-time script which is meant to just run once for existing clusters

Reply:
done